lib/max_subarray.py

```diff
@@ -2,11 +2,21 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+    n = len(nums)
+    max_sum = -10000000000
```

Comment:
👀 What if the list were made of negative numbers all below -10000000000?

We could use an impossibly small value (like `float('-inf')`) or grab a value from the array (we know that `nums[0]` at least must exist).